### PR TITLE
Add support for ignore

### PR DESCRIPTION
**Pull request by clearr**
This adds support for `git_ignore_*` functions.

### CODE REVIEW

File: ignore.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+/*
+#include <git2.h>
+*/
+import "C"
+import (
+	"runtime"
+)
+
+func (v *Repository) AddInternalIgnoreRules(rules string) error {
```
Comment (carlosmn) [inline ignore.go]: This name doesn't match the libgit2 name. It should also not be plural, as it only accepts a single rule to add.

Comment (clearr) [inline ignore.go]: [This document](https://libgit2.github.com/libgit2/#HEAD/group/ignore/git_ignore_add_rule) says that the parameter `rules` can have multiple rules in it.
Anyway, I will change it to `AddIgnoreRule(rules string)` to match libgit2.

File: ignore.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import "C"
+import (
+	"runtime"
+)
+
+func (v *Repository) AddIgnoreRule(rules string) error {
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	ret := C.git_ignore_add_rule(v.ptr, C.CString(rules))
```
Comment (carlosmn) [inline ignore.go]: This (and the one below) leaks the string. You need to keep it in a variable and free it after it's done via `C.free()`, which is usually done in a `defer`. Check out the other places in the library where we do this for examples.

**Comment by clearr** [comment]
I fixed leaks and squashed commits.

**Pull request merged**